fetch.bs

```diff
@@ -2813,6 +2854,35 @@ with a <i>CORS flag</i> and <i>recursive flag</i>, run these steps:
    <li><a lt="should response to request be blocked due to nosniff">should <var>internalResponse</var> to <var>request</var> be blocked due to nosniff</a>
   </ul>
 
+ <li>
+  <p>If <var>internalResponse</var>'s <a for=response>status</a> is <code>206</code>,
+  <var>response</var>'s <a for=response>type</a> is "<code>opaque</code>",
+  <var>internalResponse</var>'s <a for=response>range requested flag</a> is set, and
+  <var>request</var>'s <a for=request>header list</a> does not <a for="header list">contain</a>
+  `<code>Range</code>`, then:
+
+  <ol>
+   <li><p>Set <var>internalResponse</var> to a new <a for=/>response</a>.
+
+   <li><p>Set <var>response</var>'s <a>internal response</a> to <var>internalResponse</var>.
+  </ol>
+
+  <div class=note>
+   <p>Traditionally, APIs accept a ranged response even if a range wasn't requested. However, we
+   need to prevent a partial response from an earlier ranged request being provided to an API that
+   didn't make a range request.
+
+   <p>The above step prevents the following attack attack:
+
+   <p>A media element is used to request a range of a cross-origin HTML resource. Although this is
+   invalid media, a reference to a clone of the response can be retained in a service worker. This
+   can later be used as the response to a script element's fetch. If the partial response is valid
+   JavaScript (even though the whole resource is not), executing it would leak private data.
+
+   <p>The response is replaced with an "ok" response to avoid triggering error events or rejections
+   in APIs, as this would also leak information about the response.
+  </div>
+
```

Comment:
I wonder if this should copy some fields as we established in the CORB PR, to keep CSP and such working. I guess maybe not since it happens at a later point?